lib/remote/actionshandler.cpp

```diff
@@ -88,7 +89,28 @@ bool ActionsHandler::HandleRequest(
 	if (params)
 		verbose = HttpUtility::GetLastParameter(params, "verbose");
 
+	std::shared_lock wgLock{*waitGroup, std::try_to_lock};
+	if (!wgLock) {
+		HttpUtility::SendJsonError(response, params, 503, "Shutting down.");
+		return true;
+	}
+
 	for (ConfigObject::Ptr obj : objs) {
+		if (!waitGroup->IsLockable()) {
+			if (wgLock) {
+				wgLock.unlock();
+			}
+			
+			Dictionary::Ptr fail = new Dictionary({
+				{ "code", 503 },
+				{ "status", "Action skipped: Shutting down."}
+			});
+
+			results.emplace_back(std::move(fail));
```

Comment:
I'd also add "type" and "name" to the response like my other comment suggests for the other two handlers. Here, the other cases don't already do this, but with the current code, you just get a bunch of repetitions of that very same JSON without any reference to the object. So repeating it doesn't really provide any benefit.